test/e2e/reconciler/clusterworkspacedeletion/controller_test.go

```diff
@@ -227,19 +227,23 @@ func TestWorkspaceDeletionController(t *testing.T) {
 				t.Logf("Ensure workspace in the org workspace is deleted")
 				require.Eventually(t, func() bool {
 					wslist, err := server.orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().List(ctx, metav1.ListOptions{})
+					// 404 could be returned if the org workspace is deleted.
+					if apierrors.IsNotFound(err) {
+						return true
+					}
+
 					if err != nil {
 						return false
 					}
 
 					return len(wslist.Items) == 0
-				}, wait.ForeverTestTimeout, 1*time.Second)
+				}, 2*wait.ForeverTestTimeout, 1*time.Second)
```

Comment:
2 * forever = smell. Do we know what takes more than 30s here?

Reply:
I actually am thinking that handling the 404 will fix the problem, so we could start with that, and not do this 2* change now